lua/bson_lua_wrapper.c

```diff
@@ -0,0 +1,372 @@
+#include <stdlib.h>
+#include <string.h>
+ 
+#include <lua5.2/lua.h>
+#include <lua5.2/lauxlib.h>
+ 
+#include <bson_object.h>
+
+static int table_to_bson_object(lua_State *L, BsonObject *obj, char *errorMessage);
+static int table_to_bson_array(lua_State *L, BsonArray *arr, char *errorMessage);
+
+static int bson_object_to_table(lua_State *L, BsonObject *obj, char *errorMessage);
+static int bson_element_to_table(lua_State *L, BsonElement *element, char *errorMessage);
+static int bson_array_to_table(lua_State *L, BsonArray *arr, char *errorMessage);
+
+static int lbson_object_to_bytes(lua_State *L) {
+    BsonObject bson;
+    bson_object_initialize_default(&bson);
+    if (!lua_istable(L, 1)) {
```

Comment:
@jacobkeeler  lua_istable return integer, I would avoid implicit casting to bool